### DISCUSSION TITLE
[Feature]: documentation for how to build the examples

### DISCUSSION THREAD

**Discussion by eedeebee** [TheWidlarzGroup/react-native-video#4972] [Ideas]
### What problem are you trying to solve?

I tried to build your basic example app and couldn't on iOS and Android. Given there are no instructions, I wasn't sure how to do it. I tried npm (nope), yarn part of the way plus pod install for iOS but then iOS complained it could find index.js. Android wouldn't build for some other reason I've now lost, but regardless, it would be great to know how to do this.

Working examples are better than sample code.

### Proposed solution

Documentation for how to build the examples.

### Platforms

iOS, Android

### Additional context

**Possible implementation:** Technical POV how to do it

**Comment by KrzysztofMoch** [comment]
1. You clone the repo
2. You run `yarn install` in (repo) root folder
3. You go to `basic` example
4. You run `yarn install` in app folder
4.5 You run `pod install` command in app iOS folder
5. You run `yarn start` and can decide what you want to do

I have just cloned the repo and with those steps were able to successful build and run both android and iOS

**Comment by eedeebee** [comment]
Thank you!  (I would suggest adding the docs in a README or somewhere in the repo for others to find).  On Apr 14, 2024, at 2:18 PM, Olivier Bouillet ***@***.***> wrote:﻿
Closed #3648 as completed.

**Comment by abhishekjoshi-growthaxl** [comment]
Followed these steps : 

1- clone the repo
2- run yarn install in (repo) root folder
3- go to basic example
4- run yarn install in app folder
5- run pod install command in app iOS folder

It is giving this issue - 

@KrzysztofMoch 



> Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.
>  ERROR  Invariant Violation: "main" has not been registered. This can happen if:
> * Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
> * A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes

<img width="342" alt="Screenshot 2024-07-19 at 3 38 19 PM" src="https://github.com/user-attachments/assets/0731cb68-6605-42fe-a238-eb9d5af6f4bf">

**Comment by Okelm** [comment]
@abhishekjoshi-growthaxl try this: 
...
5. You run yarn start 
6. Open xCode and hit "run"

**Comment by eedeebee** [comment]
I would suggest these details belong in README somewhere instead of a github issue :)

**Comment by Okelm** [comment]
Samples are being rebuilt right now so all instructions are going to be updated

**Comment by TWG-AI** [comment]
Thanks for raising this — the team will consider it. Flagged as a potential good first issue for contributors interested in picking it up.

> I'm an AI assistant — if you're interested in contributing, check the CONTRIBUTING guide for setup steps.
